### PR TITLE
feat(state): add subscribe/unsubscribe

### PR DESCRIPTION
**Pull request by rvsia**
parts of https://github.com/data-driven-forms/react-forms/issues/705

- also implements FormSpy

### CODE REVIEW

File: packages/form-state-manager/src/types/form-spy.d.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { Subscription } from './use-subscription';
+import AnyObject from './any-object';
+
+type Children = (props: AnyObject) => React.ReactElement;
```
Comment (Hyperkid123) [inline packages/form-state-manager/src/types/form-spy.d.ts:4]: I think that we should pass specific props to the children's function. Props like value, and field metadata

**Comment by Hyperkid123** [comment]
@rvsia :arrow_forward:  :basecamp:

File: packages/form-state-manager/src/tests/form-spy.test.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    wrapper.update();
+
+    expect(render.mock.calls.length).toEqual(2);
+
+    await act(async () => {
+      managerApi().rerender(['values']);
+    });
+    wrapper.update();
+
+    expect(render.mock.calls.length).toEqual(2);
```
Comment (Hyperkid123) [inline packages/form-state-manager/src/tests/form-spy.test.js:44]: You could also check that the render function receives correct values.

Comment (rvsia) [reply packages/form-state-manager/src/tests/form-spy.test.js:44]: the check is there on line 21. (the state is not being changed in the form manager (and we don't rerender form on chanfes) so there is no need check it multipletimes)

**Pull request merged**